modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/ObjectEntry1toMObjectRelatedModelsProviderImpl.java

```diff
@@ -68,14 +76,67 @@ public void deleteRelatedModel(
 				deletionType,
 				ObjectRelationshipConstants.DELETION_TYPE_CASCADE)) {
 
+			int maxpoolsize = GetterUtil.getInteger(
+				SystemProperties.get("system.executor.service.maxpoolsize"), 2);
+
+			ExecutorService executorService = new ThreadPoolExecutor(
+				maxpoolsize, maxpoolsize);
+
+			List<Future<?>> futures = new ArrayList<>();
+
+			String name = PrincipalThreadLocal.getName();
+			PermissionChecker permissionChecker =
+				PermissionThreadLocal.getPermissionChecker();
+
 			for (ObjectEntry objectEntry : relatedModels) {
-				_objectEntryService.deleteObjectEntry(
-					objectEntry.getObjectEntryId());
+				futures.add(
+					executorService.submit(
+						() -> {
+							try {
+								PermissionThreadLocal.setPermissionChecker(
```

Comment:
@pedro-oliveira446 Please make sure all thread locals are reset if you follow my suggestion above to reuse executorService as thread pool will not reset your threadlocals and provide you a clean thread each time automatically